src/string_bytes.cc

```diff
@@ -607,6 +608,12 @@ size_t StringBytes::hex_encode(
   return dlen;
 }
 
+std::string StringBytes::hex_encode(const char* src, size_t slen) {
+  size_t dlen = slen * 2;
+  char* dst = node::UncheckedMalloc(dlen);
+  hex_encode(src, slen, dst, dlen);
+  return std::string(dst, dlen);
```

Comment:
The `UncheckedMalloc` call leaks memory :)

You should be able to avoid the extra allocation anyway by using something along the lines of 

```c++
  size_t dlen = slen * 2;
  std::string dst(dlen);
  hex_encode(src, slen, &dst[0], dlen);
  return dst;
```